src/Sulu/Bundle/AdminBundle/Resources/js/components/CircleSelection/CircleSelectionRenderer.js

```diff
@@ -0,0 +1,196 @@
+// @flow
+import React from 'react';
+import type {Normalizer, SelectionData} from './types';
+import ModifiableCircle from './ModifiableCircle';
+import PositionNormalizer from './normalizers/PositionNormalizer';
+import RoundingNormalizer from './normalizers/RoundingNormalizer';
+import SizeNormalizer from './normalizers/SizeNormalizer';
+
+type Props = {
+    containerHeight: number,
+    containerWidth: number,
+    disabled: boolean,
+    filled: boolean,
+    label?: string,
+    maxRadius?: number,
+    minRadius?: number,
+    onChange: (value: ?SelectionData) => void,
+    percentageValues: boolean,
+    resizable: boolean,
+    round: boolean,
+    value: SelectionData | typeof undefined,
+};
+
+export default class CircleSelectionRenderer extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        filled: false,
+        percentageValues: false,
+        resizable: true,
+        round: true,
+    };
+
+    handleChange = (value: ?SelectionData) => {
+        const {onChange, containerWidth, containerHeight, percentageValues} = this.props;
+
+        if (!percentageValues || !value) {
+            onChange(value);
+
+            return;
+        }
+
+        const {left, top, radius = 0} = value;
+
+        onChange({
+            left: left / containerWidth,
+            top: top / containerHeight,
+            radius: radius / containerWidth,
+        });
+    };
+
+    get value() {
```

Comment:
Because it doesn't use any mobx state

Reply:
It probably should then. If you make a component an `observer` its props will also be `observable`s, then the `computed` would make sense, wouldn't it?

Reply:
But if a prop changes, the component will also be updated even if I don't observe those props, right? Maybe I just don't understand what you mean 🙂

Reply:
I don't think so, because the `@observer` decorator will add a `shouldComponentUpdate` to the React component, that handles that. So MobX takes then care whether the component should be rerendered, not React anymore. Therefore it should then only rerender if a prop is changed that has also an effect on the `render` method. So no, changing props will not automatically mean that the component gets rerendered, that's why making this a `@computed` still makes sense.

Reply:
ah okay, now I understand 🙂